solidity/contracts/middleware/libs/OwnableMulticall.sol

```diff
@@ -36,9 +36,25 @@ contract OwnableMulticall {
         commitment = _commitment;
     }
 
-    /// @notice Deletes the commitment after it has been executed.
-    function deleteCommitment() external onlyOwner {
-        delete commitment;
+    /// @dev The calls represented by the commitment can only be executed once per commitment,
+    /// though you can submit the same commitment again after the calls have been executed.
+    function revealAndExecute(
+        CallLib.Call[] calldata calls,
+        bytes32 salt
+    ) external payable returns (bytes32 executedCommitment) {
+        // If there is no active commitment, do nothing.
+        if (commitment == bytes32(0)) return bytes32(0);
```

Comment:
I expected you to move the
```
require(
            ica.commitment() == msgCommitment,
            "Commitment ISM: Invalid Commitment"
        );
```
block here
we shouldnt need to call into this contract multiple times from the ISM